conans/test/functional/compression/compression_test.py

```diff
@@ -0,0 +1,443 @@
+        self.c1 = TestClient()
+        self.c2 = TestClient()
+        self.c1.save({}, clean_first=True)
+        self.c2.save({}, clean_first=True)
+
+    def tearDown(self):
+        # common for all tests below
+        self.compressor.compress(self.c1.current_folder)
+        self.compressor.decompress(self.c2.current_folder)
+
+        self.compare_dirs(self.c1.current_folder, self.c2.current_folder)
+
+    def test_basic(self):
+        self.c1.save(self.gen_files(client=self.c1, count=10, level=1, filesize=10*1024),
+                     clean_first=True)
+
+    # simple valid file symbolic link
+    @unittest.skipUnless(symlinks_supported(), "requires symlinks")
+    def test_simple_file_symlink(self):
+        files = self.gen_files(client=self.c1)
+
+        with chdir(self.c1.current_folder):
+            create_symlink(list(files.keys())[0], self.gen_filename())
+
+    # simple valid directory symbolic link
+    @unittest.skipUnless(symlinks_supported(), "requires symlinks")
+    def test_simple_dir_symlink(self):
+        files = self.gen_files(client=self.c1, level=1)
+
+        with chdir(self.c1.current_folder):
+            filename = list(files.keys())[0]
+            dirname = os.path.dirname(filename)
+            create_symlink(dirname, self.gen_filename())
+
+    # simple valid file symbolic link (absolute)
+    @unittest.skipUnless(symlinks_supported(), "requires symlinks")
+    def test_abs_symlink(self):
+        files = self.gen_files(client=self.c1)
+
+        with chdir(self.c1.current_folder):
+            create_symlink(os.path.abspath(list(files.keys())[0]), self.gen_filename())
+
+    # valid relative symbolic link to the parent directory like "../A/B"
+    @unittest.skipUnless(symlinks_supported(), "requires symlinks")
+    def test_relative_symlink(self):
+        files = self.gen_files(client=self.c1, level=1, count=2)
+
+        with chdir(self.c1.current_folder):
+            a = list(files.keys())[0]
+            b = list(files.keys())[1]
+            os.unlink(b)
+            create_symlink(os.path.relpath(a, os.path.dirname(b)) ,b)
+
+    # symbolic link pointing to the non-existing file
+    @unittest.skipUnless(symlinks_supported(), "requires symlinks")
+    def test_broken_symlink(self):
+        with chdir(self.c1.current_folder):
+            create_symlink(self.gen_filename(), self.gen_filename())
+
+    # symbolink link pointing outside of the archive (e.g. system location like "/dev/null")
+    @unittest.skipUnless(symlinks_supported(), "requires symlinks")
+    def test_symlink_outside(self):
+        self.c3 = TestClient()
+        files = self.gen_files(client=self.c3)
+
+        with chdir(self.c1.current_folder):
+            create_symlink(os.path.join(self.c3.current_folder, list(files.keys())[0]),
+                           self.gen_filename())
+
+    # cyclic reference A -> B, B -> A
+    @unittest.skipUnless(symlinks_supported(), "requires symlinks")
```

Comment:
I cannot find transitive symlinks A->B->C. Is there any test for this case?

Reply:
okay, let's add this one as well

Reply:
added `test_transitive_symlinks`